File: FoodOrderingApp/Public.DTO/v1/OrderItem.cs
using Base.Contracts.Domain;

namespace Public.DTO.v1;

public class OrderItem : IDomainEntityId
{
    public Guid Id { get; set; }

    public uint Quantity { get; set; }

    public Guid ProductId { get; set; }
    public ProductPure? Product { get; set; }

    public Guid PriceId { get; set; }
    public Price? Price { get; set; }

    public decimal PriceValue { get; set; }

    public Guid OrderId { get; set; }
}

public class OrderItemRequest
{
    public Guid? Id { get; set; }

    public uint Quantity { get; set; }
    public Guid ProductId { get; set; }
    public Guid PriceId { get; set; }
    public Guid? OrderId { get; set; }
}

File: food-ordering-system/src/app/shared/models/OrderItem.ts
import {Product} from "./Product";
import {Price} from "./Price";

export type OrderItem = {
  id: string;

  quantity: number;
  productId: string;
  product: Product;

  priceId: string;
  price: Price;

  priceValue: string;
  orderId: string;
}

export type OrderItemRequest = {
  id?: string;
  quantity: number;
  productId: string;
  priceId: string;
  orderId?: string;
}
